test/integration/server.h

```diff
@@ -462,6 +462,10 @@ class IntegrationTestServer : public Logger::Loggable<Logger::Id::testing>,
     notifyingStatsAllocator().waitForCounterExists(name);
   }
 
+  void waitForGaugeDestroyed(const std::string& name) override {
+    ASSERT_TRUE(TestUtility::waitForGaugeDestroyed(statStore(), name, time_system_));
```

Comment:
Open an issue and add TODO(#issue) to add the support in notifyingStatsAllocator for gauges.

I don't think replicating the counter-notification infra will be at all difficult; just copy/paste/tweak the counter support. You'd also need to add support for signaling the mutex when a gauge is removed. Also not difficult, but it's a new flavor of hook.

Reply:
done